File: 9 - Special Pythagorean Triplet.rb
#A Pythagorean triplet is a set of three natural numbers,
#a < b < c, for which,
#a2 + b2 = c2

#For example, 32 + 42 = 9 + 16 = 25 = 52.

#There exists exactly one Pythagorean triplet for which a + b + c = 1000.
#Find the product abc.

sum = 0
a = 0
b = 1
c = 0

while sum != 1000
  a += 1
  b = a
  while a + 2*b < 1000
    b += 1
    c = Math.sqrt(a*a + b*b)
    if c % 1 == 0         #check that sqrt returned a whole number
      c = c.to_i
      sum = a+b+c
      break if sum == 1000
    end
  end
end

puts a*b*c

File: python/9_Special_Pythagorean_Triplet.py
#!/usr/bin/python
"""Find the only Pythagorean triplet, {a, b, c}, for which a + b + c = 1000.
   Find the product of abc.	
"""

from math import sqrt

if __name__ == "__main__":

	for a in range(1,500,1):
		for b in range(a+1,1000,1):

			if sqrt((a**2)+(b**2)) == int(sqrt((a**2)+(b**2))):
				if (a+b+int(sqrt(((a**2)+(b**2)))))==1000:
					print a*b*int(sqrt((a**2)+(b**2)))
					exit(0)
				
